pkg/cmd/secret/set/set.go

```diff
@@ -0,0 +1,216 @@
+					"--repos flag required when --visibility='selected'")}
+			}
+
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return setRun(opts)
+		},
+	}
+	cmd.Flags().StringVar(&opts.OrgName, "org", "", "List secrets for an organization")
+	cmd.Flags().Lookup("org").NoOptDefVal = "@owner"
+	cmd.Flags().StringVarP(&opts.Visibility, "visibility", "v", "private", "Set visibility for an organization secret: `all`, `private`, or `selected`")
+	cmd.Flags().StringSliceVarP(&opts.RepositoryNames, "repos", "r", []string{}, "List of repository names for `selected` visibility")
+	cmd.Flags().StringVarP(&opts.Body, "body", "b", "-", "Provide either a literal string or a file path; prepend file paths with an @. Reads from STDIN if not provided.")
+
+	return cmd
+}
+
+func setRun(opts *SetOptions) error {
+	body, err := getBody(opts)
+	if err != nil {
+		return fmt.Errorf("did not understand secret body: %w", err)
+	}
+
+	c, err := opts.HttpClient()
+	if err != nil {
+		return fmt.Errorf("could not create http client: %w", err)
+	}
+	client := api.NewClientFromHTTP(c)
+
+	var baseRepo ghrepo.Interface
+	if opts.OrgName == "" || opts.OrgName == "@owner" {
+		baseRepo, err = opts.BaseRepo()
+		if err != nil {
+			return fmt.Errorf("could not determine base repo: %w", err)
+		}
+	}
+
+	host := ghinstance.OverridableDefault()
+	if opts.OrgName == "@owner" {
+		opts.OrgName = baseRepo.RepoOwner()
+		host = baseRepo.RepoHost()
+	}
+
+	var pk *PubKey
+	if opts.OrgName != "" {
+		pk, err = getOrgPublicKey(client, host, opts.OrgName)
+	} else {
+		pk, err = getRepoPubKey(client, baseRepo)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to fetch public key: %w", err)
+	}
+
+	eBody, err := box.SealAnonymous(nil, body, &pk.Raw, opts.RandomOverride)
+	if err != nil {
+		return fmt.Errorf("failed to encrypt body: %w", err)
+	}
+
+	encoded := base64.StdEncoding.EncodeToString(eBody)
+
+	if opts.OrgName != "" {
+		err = putOrgSecret(client, pk, host, *opts, encoded)
+	} else {
+		err = putRepoSecret(client, pk, baseRepo, opts.SecretName, encoded)
+	}
+	if err != nil {
+		return fmt.Errorf("failed to set secret: %w", err)
+	}
+
+	if opts.IO.IsStdoutTTY() {
+		cs := opts.IO.ColorScheme()
+		fmt.Fprintf(opts.IO.Out, "%s Set secret %s\n", cs.SuccessIcon(), opts.SecretName)
+	}
+
+	return nil
+}
+
+func validSecretName(name string) error {
+	if name == "" {
+		return errors.New("secret name cannot be blank")
+	}
+
+	if strings.HasPrefix(name, "GITHUB_") {
+		return errors.New("secret name cannot begin with GITHUB_")
+	}
+
+	leadingNumber := regexp.MustCompile(`^[0-9]`)
+	if leadingNumber.MatchString(name) {
+		return errors.New("secret name cannot start with a number")
+	}
+
+	validChars := regexp.MustCompile(`^([0-9]|[a-z]|[A-Z]|_)+$`)
+	if !validChars.MatchString(name) {
+		return errors.New("secret name can only contain letters, numbers, and _")
+	}
+
+	return nil
+}
+
+func getBody(opts *SetOptions) (body []byte, err error) {
+	if opts.Body == "-" {
```

Comment:
Usually, it's convention that commands accept the `-` value in place of a filename to signal reading from stdin instead of a file. Since the `-b` flag only accepts a filename after a `@`, would it make more sense that this condition is captured as `if opts.Body == "@-"`?

For example, from `gh help api`:
```
The '--field, -F' flag:
...
- if the value starts with "@", the rest of the value is interpreted as a
  filename to read the value from. Pass "-" to read from standard input.
```

Therefore, a way to pass stdin to that flag would be `-F query=@-`.